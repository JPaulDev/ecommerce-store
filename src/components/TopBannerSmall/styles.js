import styled from 'styled-components';

export const Container = styled.div`
  align-items: center;
  background-color: var(--color-grey-6);
  display: flex;
  justify-content: space-around;
  min-height: 55px;
  position: relative;
`;

export const InnerContainer = styled.div`
  align-items: center;
  background-color: var(--color-grey-6);
  column-gap: 15px;
  display: flex;
  justify-content: center;
  opacity: ${({ isActive }) => (isActive ? '1' : '0')};
  z-index: ${({ isActive }) => (isActive ? '10' : '0')};

  @media screen and (max-width: 999px) {
    min-width: 285px;
    transition: opacity 1s ease-in-out;
    position: absolute;
  }

  @media screen and (min-width: 1000px) {
    opacity: 1;
  }
`;

export const Image = styled.img`
  height: auto;
  width: ${({ displayWidth }) => displayWidth};
`;

export const Text = styled.span`
  color: var(--color-grey-2);
  font-size: 0.7rem;
`;

export const BoldText = styled.strong`
  display: block;
  font-size: 0.87rem;
`;
