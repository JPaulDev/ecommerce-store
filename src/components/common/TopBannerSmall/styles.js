import styled from 'styled-components';

export const Container = styled.div`
  align-items: center;
  background-color: var(--color-grey-6);
  display: flex;
  justify-content: space-around;
  min-height: 55px;
  position: relative;
`;

export const InnerContainer = styled.div`
  align-items: center;
  background-color: var(--color-grey-6);
  column-gap: 15px;
  display: flex;
  justify-content: center;
  opacity: ${({ isActive }) => (isActive ? '1' : '0')};
  z-index: ${({ isActive }) => (isActive ? '10' : '0')};

  @media screen and (max-width: 999px) {
    min-width: 285px;
    transition: opacity 1s ease-in-out ${({ isActive }) => isActive && '1s'};
    position: absolute;
  }

  @media screen and (min-width: 1000px) {
    opacity: 1;
  }
`;

export const Text = styled.div`
  color: var(--color-grey-2);
  font-size: 0.7rem;

  strong {
    display: block;
    font-size: 0.87rem;
  }
`;
