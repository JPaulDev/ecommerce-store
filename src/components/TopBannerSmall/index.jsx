import { useState, useEffect } from 'react';
import useMediaQuery from '../../hooks/useMediaQuery';
import trophy from '../../assets/images/top-banner-small/trophy-logo.webp';
import trustpilot from '../../assets/images/top-banner-small/trustpilot-logo.webp';
import paypal from '../../assets/images/top-banner-small/paypal-logo.webp';
import * as Styled from './styles';

const content = [
  {
    image: trophy,
    width: 119,
    height: 55,
    displayWidth: '60px',
    boldText: 'Award winning',
    text: 'Systems and customer services',
  },
  {
    image: trustpilot,
    width: 121,
    height: 56,
    displayWidth: '73px',
    boldText: '4.8 stars on Trustpilot',
    text: 'From over 30,000 customer reviews',
  },
  {
    image: paypal,
    width: 167,
    height: 56,
    displayWidth: '80px',
    boldText: '0% interest up to 4 months',
    text: 'Minimum spend applies from Â£99',
  },
];

export default function TopBannerSmall() {
  const [slideIndex, setSlideIndex] = useState(1);
  const isMatch = useMediaQuery('(min-width: 1000px)');

  useEffect(() => {
    let timeout;
    const nextSlide = (slideIndex + 1) % content.length;

    if (!isMatch) {
      timeout = setTimeout(() => {
        setSlideIndex(nextSlide);
      }, 5000);
    }

    return () => clearTimeout(timeout);
  }, [isMatch, slideIndex]);

  return (
    <Styled.Container>
      {content.map((item, index) => (
        <Styled.InnerContainer
          key={item.text}
          isActive={index === slideIndex}
          data-testid={index}
        >
          <Styled.Image
            src={item.image}
            width={item.width}
            height={item.height}
            displayWidth={item.displayWidth}
            alt=""
          />
          <Styled.Text>
            <Styled.BoldText>{item.boldText}</Styled.BoldText>
            {item.text}
          </Styled.Text>
        </Styled.InnerContainer>
      ))}
    </Styled.Container>
  );
}
