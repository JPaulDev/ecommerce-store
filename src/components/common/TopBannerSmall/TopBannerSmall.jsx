import Image from 'next/future/image';
import { useState, useEffect } from 'react';
import trophy from '../../../../public/images/top-banner-small/trophy-logo.webp';
import trustpilot from '../../../../public/images/top-banner-small/trustpilot-logo.webp';
import paypal from '../../../../public/images/top-banner-small/paypal-logo.webp';
import useMediaQuery from '../../../lib/hooks/useMediaQuery';
import * as Styled from './styles';

const slides = [
  {
    image: trophy,
    boldText: 'Award winning',
    text: 'Systems and customer services',
  },
  {
    image: trustpilot,
    boldText: '4.8 stars on Trustpilot',
    text: 'From over 30,000 customer reviews',
  },
  {
    image: paypal,
    boldText: '0% interest up to 4 months',
    text: 'Minimum spend applies from Â£99',
  },
];

export default function TopBannerSmall() {
  const [slideIndex, setSlideIndex] = useState(1);
  const isMatch = useMediaQuery('(max-width: 999px)');

  useEffect(() => {
    let timeout;
    const nextSlide = (slideIndex + 1) % slides.length;

    if (isMatch) {
      timeout = setTimeout(() => {
        setSlideIndex(nextSlide);
      }, 5000);
    }

    return () => {
      clearTimeout(timeout);
    };
  }, [isMatch, slideIndex]);

  return (
    <Styled.Container>
      {slides.map((item, index) => (
        <Styled.InnerContainer
          key={item.text}
          isActive={index === slideIndex}
          data-testid={index}
        >
          <Image src={item.image} alt="" quality={100} />
          <Styled.Text>
            <strong>{item.boldText}</strong>
            {item.text}
          </Styled.Text>
        </Styled.InnerContainer>
      ))}
    </Styled.Container>
  );
}
