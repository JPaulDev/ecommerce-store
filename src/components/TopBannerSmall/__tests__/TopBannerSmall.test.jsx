import { render, screen, act } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TopBannerSmall from '../index';

describe('TopBannerSmall component', () => {
  it('renders a banner with 3 icons and text next to each one', () => {
    const tree = renderer.create(<TopBannerSmall />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('TopBannerSmall component at a screen width of 999px and below', () => {
  it('cycles through and displays each slide', () => {
    window.resizeTo(999, 769);
    jest.useFakeTimers();
    render(<TopBannerSmall />);

    act(() => jest.runAllTimers());
    expect(screen.getByTestId('0')).toHaveStyle('opacity: 0');
    expect(screen.getByTestId('1')).toHaveStyle('opacity: 0');
    expect(screen.getByTestId('2')).toHaveStyle('opacity: 1');

    act(() => jest.runAllTimers());
    expect(screen.getByTestId('0')).toHaveStyle('opacity: 1');
    expect(screen.getByTestId('1')).toHaveStyle('opacity: 0');
    expect(screen.getByTestId('2')).toHaveStyle('opacity: 0');
  });
});
