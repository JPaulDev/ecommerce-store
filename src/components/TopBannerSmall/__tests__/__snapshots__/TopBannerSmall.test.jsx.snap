// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBannerSmall component renders a banner with 3 icons and text next to each one 1`] = `
<div
  className="sc-bczRLJ cKrcLx"
>
  <div
    className="sc-gsnTZi ehuJPo"
    data-testid={0}
  >
    <img
      alt=""
      height="28px"
      src="trophy-logo.webp"
      width="60px"
    />
    <span
      className="sc-dkzDqf eDHhsi"
    >
      <strong
        className="sc-hKMtZM hPKlJ"
      >
        Award winning
      </strong>
      Systems and customer services
    </span>
  </div>
  <div
    className="sc-gsnTZi hdgHuI"
    data-testid={1}
  >
    <img
      alt=""
      height="34px"
      src="trustpilot-logo.webp"
      width="73px"
    />
    <span
      className="sc-dkzDqf eDHhsi"
    >
      <strong
        className="sc-hKMtZM hPKlJ"
      >
        4.8 stars on Trustpilot
      </strong>
      From over 30,000 customer reviews
    </span>
  </div>
  <div
    className="sc-gsnTZi ehuJPo"
    data-testid={2}
  >
    <img
      alt=""
      height="27px"
      src="paypal-logo.webp"
      width="80px"
    />
    <span
      className="sc-dkzDqf eDHhsi"
    >
      <strong
        className="sc-hKMtZM hPKlJ"
      >
        0% interest up to 4 months
      </strong>
      Minimum spend applies from Â£99
    </span>
  </div>
</div>
`;
